tvtk/pyface/ui/qt4/QVTKRenderWindowInteractor.py

```diff
@@ -424,11 +432,37 @@ def keyReleaseEvent(self, ev):
         self._Iren.KeyReleaseEvent()
 
     def wheelEvent(self, ev):
+        """ Reimplemented to work around scrolling bug in Mac.
+
+        Work around https://bugreports.qt-project.org/browse/QTBUG-22269.
+        Accumulate wheel events that are within a period of 25ms into a single
+        event.  Changes in buttons or modifiers, while a scroll is going on,
+        are not handled, since they seem to be too much of a corner case to be
+        worth handling.
+        """
+        self.wheel_accumulator += ev.delta()
+        self._saved_wheel_event_info = (
```

Comment:
I think you should accumulate the events only when they are similar. I.e. check that atleast the buttons, modifiers and orientation of the new event match the properties of events accumulated upto now, otherwise diapatch the accumulated one and start accumulating the new event from delta=0.